manager/role_manager.go

```diff
@@ -62,13 +82,29 @@ func (rm *roleManager) Run(ctx context.Context) {
 	if err != nil {
 		log.G(ctx).WithError(err).Error("failed to check nodes for role changes")
 	} else {
+		// Put all members in the removed list first - if the node object exists
+		// for this member, then it is deleted from the removed list.  This is safe
+		// because a node object is created when a node first requests TLS certificates
+		// to be able to communicate with other raft members, so the node object
+		// will exist before the raft membership exists.
+		for _, member := range rm.raft.GetMemberlist() {
+			rm.removed[member.NodeID] = member
+		}
 		for _, node := range nodes {
+			// if the node exists, we don't want it removed from the raft membership cluster
+			// necessarily
+			delete(rm.removed, node.ID)
+
+			// reconcile each existing node
 			rm.pending[node.ID] = node
 			rm.reconcileRole(ctx, node)
 		}
-		if len(rm.pending) != 0 {
-			ticker = time.NewTicker(roleReconcileInterval)
-			tickerCh = ticker.C
+		for _, removed := range rm.removed {
```

Comment:
Please add a comment.

Reply:
Could you clarify what kind of comment would make sense here?